internal/oci/container_linux.go

```diff
@@ -41,40 +42,47 @@ func (c *Container) SeccompProfilePath() string {
 	return c.seccompProfilePath
 }
 
-// getPidStartTime reads the kernel's /proc entry for stime for PID.
+// getPidData reads the kernel's /proc entry for various data.
 // inspiration for this function came from https://github.com/containers/psgo/blob/master/internal/proc/stat.go
 // some credit goes to the psgo authors
-func getPidStartTime(pid int) (string, error) {
-	return GetPidStartTimeFromFile(fmt.Sprintf("/proc/%d/stat", pid))
+func getPidData(pid int) (*StatData, error) {
+	return getPidStatData(fmt.Sprintf("/proc/%d/stat", pid))
 }
 
-// GetPidStartTime reads a file as if it were a /proc/$pid/stat file, looking for stime for PID.
+// GetPidStartTimeFromFile reads a file as if it were a /proc/$pid/stat file, looking for stime for PID.
 // It is abstracted out to allow for unit testing
 func GetPidStartTimeFromFile(file string) (string, error) {
+	data, err := getPidStatData(file)
+	if err != nil {
+		return "", err
+	}
+	return data.StartTime, nil
+}
+
+func getPidStatData(file string) (*StatData, error) {
 	data, err := os.ReadFile(file)
```

Comment:
Some ideas for small optimisations that we can do here if we want to.

1) We can use something that won't call `fstat()` on the file, which does not work for procfs/sysfs anyway
2) We can only request for a page of data (so typically a 4 KiB on x86_64) to be read at once

The (2) is because procfs/sysfs (contrary to debugfs, etc.) only serves a page worth of data.

This can reduce the number of allocations a little bit per:

```console
$ go test -v -bench=. -benchmem 
goos: linux
goarch: amd64
pkg: test-procfs
cpu: AMD Ryzen 9 6900HX with Radeon Graphics        
BenchmarkA
BenchmarkA-16    	  256002	      4549 ns/op	    2160 B/op	       8 allocs/op
BenchmarkB
BenchmarkB-16    	  284290	      4327 ns/op	    1976 B/op	       8 allocs/op
PASS
ok  	test-procfs	3.074s
```

The `os.ReadFile()` call to `fstat()` will only return `0` (or 4 KiB; as it would for most procfs/sysfs objects) per:

```console
fstat(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
``` 

There is also a potential to leverage `fmt.Fscan()` over a bytes buffer in lieu of `strings.Split()` to potentially further reduce memory allocation.